### PR TITLE
CFY-7697 support uploading wagon with yaml

### PR DESCRIPTION
**Pull request by dusking**

### CODE REVIEW

File: rest-service/manager_rest/utils.py [hunk 1]
```diff
@@ -228,3 +231,30 @@ def current_tenant():
...
     g.current_tenant = tenant
+
+
+def unzip(archive, destination=None, logger=None):
+    if not destination:
+        destination = tempfile.mkdtemp()
+    if logger:
+        logger.debug('Extracting zip {0} to {1}...'.
+                     format(archive, destination))
+    with closing(zipfile.ZipFile(archive, 'r')) as zip_file:
```
Comment (tehasdf) [inline rest-service/manager_rest/utils.py]: the `closing` is needed? `ZipFile` should work with `with` by itself

File: rest-service/manager_rest/upload_manager.py [hunk 2]
```diff
@@ -614,10 +627,22 @@ def _prepare_and_process_doc(self,
...
+        # support previous implementation
+        wagon_target_path = archive_target_path
+
+        # handle the archive_target_path, which may be zip or wagon
+        if zipfile.is_zipfile(archive_target_path):
+            archive_name = unzip(archive_target_path,
+                                 logger=current_app.logger)
+            os.remove(archive_target_path)
+            shutil.move(archive_name, archive_target_path)
+            wagon_target_path, _ = self._verify_archive(archive_target_path)
```
Comment (tehasdf) [inline rest-service/manager_rest/upload_manager.py]: add a comment about why the returned `yamls[0]` can be ignored?

Comment (dusking) [reply rest-service/manager_rest/upload_manager.py]: no special reason, I just don't use it.. :)

File: rest-service/manager_rest/test/endpoints/test_plugins.py [hunk 3]
```diff
@@ -215,15 +215,17 @@ def raises(*args, **kwargs):
           client_max_version=base_test.LATEST_API_VERSION)
     def test_plugin_upload_progress(self):
         tmp_file_path = self.create_wheel('wagon', '0.6.1')
-        total_size = os.path.getsize(tmp_file_path)
+        yaml_path = self.get_blueprint_path('mock_blueprint/plugin.yaml')
```
Comment (yarivgiga) [inline rest-service/manager_rest/test/endpoints/test_plugins.py]: same comment...

File: rest-service/manager_rest/test/endpoints/test_plugins.py [hunk 4]
```diff
@@ -232,10 +234,12 @@ def test_plugin_upload_progress(self):
           client_max_version=base_test.LATEST_API_VERSION)
     def test_plugin_download_progress(self):
         tmp_file_path = self.create_wheel('wagon', '0.6.1')
+        yaml_path = self.get_blueprint_path('mock_blueprint/plugin.yaml')
```
Comment (yarivgiga) [inline rest-service/manager_rest/test/endpoints/test_plugins.py]: same...

File: rest-service/manager_rest/test/infrastructure/base_list_test.py [hunk 5]
```diff
@@ -55,7 +55,9 @@ def _put_n_plugins(self, number_of_plugins):
                 f.write('from setuptools import setup\n')
                 f.write('setup(name="some-package", version={0})'.format(i))
             plugin_path = wagon.create(tmpdir, archive_destination_dir=tmpdir)
-            self.post_file('/plugins', plugin_path)
+            yaml_path = self.get_blueprint_path('mock_blueprint/plugin.yaml')
```
Comment (yarivgiga) [inline rest-service/manager_rest/test/infrastructure/base_list_test.py]: ...

File: rest-service/manager_rest/utils.py [hunk 6]
```diff
@@ -228,3 +230,30 @@ def current_tenant():
...
+        logger.debug('Extracting zip {0} to {1}...'.
+                     format(archive, destination))
+    with zipfile.ZipFile(archive, 'r') as zip_file:
+        zip_file.extractall(destination)
+    return destination
+
+
+def files_in_folder(folder, name_pattern='*'):
+    files = []
+    for item in os.listdir(folder):
```
Comment (yarivgiga) [inline rest-service/manager_rest/utils.py]: note that listdir returns both files and dirs inside the given dir.
so this func may return dirs that match the pattern as well.
consider add is_file() check.

File: rest-service/manager_rest/utils.py [hunk 7]
```diff
@@ -228,3 +230,30 @@ def current_tenant():
...
+        destination = tempfile.mkdtemp()
+    if logger:
+        logger.debug('Extracting zip {0} to {1}...'.
+                     format(archive, destination))
+    with zipfile.ZipFile(archive, 'r') as zip_file:
+        zip_file.extractall(destination)
+    return destination
+
+
+def files_in_folder(folder, name_pattern='*'):
```
Comment (tehasdf) [inline rest-service/manager_rest/utils.py]: this could also possibly be replaced by the stdlib `glob.glob`

**Pull request merged**